### DISCUSSION TITLE
Trying to run this on unraid which variables are needed

### DISCUSSION THREAD

**Discussion by Diondk** [ghostfolio/ghostfolio#923] [Q&A]
Hi All,

I am trying to get this to run on unraid but its not able to start, can you tell me which variables i need to add to unraid docker to get it up and running. 

For example i have an issue with the ports now:
While i have a variable added to the container that shows port: 3333

`(node:1) UnhandledPromiseRejectionWarning: RangeError [ERR_SOCKET_BAD_PORT]: Port should be >= 0 and < 65536. Received NaN.
at new NodeError (internal/errors.js:322:7)
at validatePort (internal/validators.js:216:11)
at lookupAndConnect (net.js:1013:5)
at Socket.connect (net.js:989:5)
at Object.connect (net.js:201:17)
at /ghostfolio/apps/api/node_modules/ioredis/built/connectors/StandaloneConnector.js:58:45
at processTicksAndRejections (internal/process/task_queues.js:77:11)
(Use `node --trace-warnings ...` to show where the warning was created)

(node:1) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)

(node:1) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
events.js:377
throw er; // Unhandled 'error' event

^

Error: connect ECONNREFUSED 127.0.0.1:6379

at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1159:16)
Emitted 'error' event on RedisClient instance at:

at RedisClient.on_error (/ghostfolio/apps/api/node_modules/redis/index.js:342:14)
at Socket.<anonymous> (/ghostfolio/apps/api/node_modules/redis/index.js:223:14)
at Socket.emit (events.js:400:28)
at emitErrorNT (internal/streams/destroy.js:106:8)
at emitErrorCloseNT (internal/streams/destroy.js:74:3)
at processTicksAndRejections (internal/process/task_queues.js:82:21) {
errno: -111,
code: 'ECONNREFUSED',
syscall: 'connect',
address: '127.0.0.1',
port: 6379
}`

**Comment by dtslvr** [comment]
Hi @Diondk

It looks like the redis port is invalid:
```
Port should be >= 0 and < 65536 ... at /ghostfolio/apps/api/node_modules/ioredis/built/connectors/StandaloneConnector.js
```

You probably need to set `REDIS_HOST` and `REDIS_PORT`. Here you can find a sample with all environment variables: [.env](https://github.com/ghostfolio/ghostfolio/blob/main/.env)

**Comment by dtslvr** [comment]
→ Progress of the Unraid template: https://github.com/ghostfolio/ghostfolio/issues/928#issuecomment-1133764345

**Comment by dtslvr** [comment]
→ Link to Unraid instructions: https://github.com/ghostfolio/ghostfolio#run-with-unraid-self-hosting
